internal/cmd/autostart.go

```diff
@@ -0,0 +1,99 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/coder-cli/coder-sdk"
+	"cdr.dev/coder-cli/internal/x/xcobra"
+	"cdr.dev/coder-cli/pkg/clog"
+)
+
+func autostartCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "autostart",
+		Short: "Interact with Coder autostart settings",
+		Long:  "Perform operations on the Coder autostart settings owned by the active user.",
+	}
+
+	cmd.AddCommand(getAutostartCmd(), setAutostartCmd())
+
+	return cmd
+}
+
+func getAutostartCmd() *cobra.Command {
+	var outputFmt string
+
+	cmd := &cobra.Command{
+		Use:   "get",
+		Short: "get autostart settings owned by the active user",
+		Long:  "Get Coder autostart settings owned by the active user.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			client, err := newClient(ctx)
+			if err != nil {
+				return err
+			}
+
+			as, err := client.MyAutoStart(ctx)
+			if err != nil {
+				return err
+			}
+
+			switch outputFmt {
+			case humanOutput:
+				fmt.Printf("AutoStart Time:\t%v\n", as.AutoStart)
+			case jsonOutput:
+				err := json.NewEncoder(os.Stdout).Encode(as)
+				if err != nil {
+					return xerrors.Errorf("write autostart as JSON: %w", err)
+				}
+			default:
+				return xerrors.Errorf("unknown --output value %q", outputFmt)
+			}
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&outputFmt, "output", "o", humanOutput, "human | json")
+
+	return cmd
+}
+
+func setAutostartCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "set [time]",
+		Short: "set autostart settings owned by the active user",
+		Args:  xcobra.ExactArgs(1),
+		Long:  "Set Coder autostart settings owned by the active user. The supplied time is in HH:mm:ss format and assumes UTC.",
+		Example: `# Set my autostart preference to 10:00:00 AM UTC
+coder autostart set 10:00:00`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			client, err := newClient(ctx)
+			if err != nil {
+				return err
+			}
+			t, err := time.Parse("15:04:05", args[0])
+			if err != nil {
+				return err
+			}
+			updateReq := &coder.AutoStartRequest{
+				AutoStart: t.Format(time.RFC3339),
+			}
+			_, err = client.UpdateAutoStart(ctx, *updateReq)
+			if err != nil {
+				return xerrors.Errorf("update autostart: %w", err)
```

Comment:
how do we pick the timezone here?

Reply:
https://github.com/cdr/coder-cli/pull/236/commits/bcb983046de2720621de4ad21f1c8da05305c0f7 I think that's the intended use case?

Reply:
Yea, so I'm a little unfamiliar with some of the tools available to me, but I thought as a starting point we could set the limitation that it's parsed in UTC. The backend expects/uses UTC, and the web client just does that conversion from locale -> UTC. I'm less sure how to do that in a CLI environment.

> In the absence of a time zone indicator, Parse returns a time in UTC.

The next thought was great, maybe we could add a format string that allows an optional timezone, but I wasn't sure how to implement it properly (was referring to these standard layouts: https://yourbasic.org/golang/format-parse-string-time-date-example/

I wanted to avoid the user having to include a date, since the date itself is meaningless: we only care about the time.

If there's a clear way to do this, I'd think ideally we either have a flexible layout string or a flag for the timezone

```
coder autostart set 15:30:00 --timezone=[timezone]
```


Definitely would appreciate help on this one from someone who's more familiar with this side of things. If not, I don't think it's a show stopper to specify the time expects UTC for now

Reply:
I wonder if we can use `--timezone` flag and default to UTC?

Reply:
Honestly I'm not apposed to keeping the actual setting of the auto-start time to dashboard-only. Then we keep the boolean flag for env creating in the CLI.

Reply:
Fair, I'll remove this